src/server-glue/webresource-handlers/S3Handler.ts

```diff
@@ -0,0 +1,105 @@
+import { optionalVar, requiredVar } from '@balena/env-parsing';
+import {
+	IncomingFile,
+	UploadResponse,
+	WebResourceHandler,
+} from '../webresource-handler';
+import {
+	S3Client,
+	S3ClientConfig,
+	DeleteObjectCommand,
+} from '@aws-sdk/client-s3';
+import { Upload } from '@aws-sdk/lib-storage';
+import { v4 as uuidv4 } from 'uuid';
+
+export interface S3HandlerConfig {
+	bucket?: string;
+	region?: string;
+	accessKeyId?: string;
+	secretAccessKey?: string;
+	endpoint?: string;
+}
+
+export class S3Handler implements WebResourceHandler {
+	private readonly bucket: string;
+	private readonly region: string;
+	private readonly accessKeyId: string;
+	private readonly secretAccessKey: string;
+	private readonly endpoint: string;
```

Comment:
isn't this actually an instance of the S3HandlerConfig?

Reply:
Yep, refactored most of this class.